backup.sh

```diff
@@ -255,15 +282,17 @@ then
   sleep 2
 
   # Compress
-  cecho "Compressing & encrypting data - this will take a while."
-  # -p: encrypt backup
-  # -mhe=on: encrypt headers (metadata)
-  # -mx=9: ultra compression
-  # -bb3: verbose logging
-  # -sdel: delete files after compression
-  # The undefined variable is set by the user
-  declare backup_archive="$archive_path/linux-android-backup-$(date +%m-%d-%Y-%H-%M-%S).7z"
-  retry 5 7z a -p$archive_password -mhe=on -mx=9 -bb3 -sdel $backup_archive backup-tmp/*
+  if [ "$compression_level" != 0 ]; then
+    cecho "Compressing & encrypting data - this will take a while."
+    # -p: encrypt backup
+    # -mhe=on: encrypt headers (metadata)
+    # -mx=9: ultra compression
+    # -bb3: verbose logging
+    # -sdel: delete files after compression
+    # The undefined variable is set by the user
+    declare backup_archive="$archive_path/linux-android-backup-$(date +%m-%d-%Y-%H-%M-%S).7z"
```

Comment:
in order to ensure compatibility with hooks, `backup_archive` should be defined regardless of whether the user's using compression or not.

https://github.com/mrrfv/linux-android-backup/blob/0fa07936f958801d027d29758a08235d8f9c6acd/README.md?plain=1#L81